upgrade/upgrade-scripts/upgrade-container

```diff
@@ -507,13 +507,30 @@ function migrate_dir() {
 	fi
 }
 
+#
+# Preserve the persistent service state of the service named as an argument.
```

Comment:
I don't know if this would be better, but we could perhaps change this function to only enable services if they're currently enabled. This would then rely on the default value of all services, which would start out as masked or disabled (at least for the services we manipulate), and only enable them when needed. This might make this function's implementation much more simple.

With that said, I'm OK with what you have here too.

Reply:
Now that I better understand how we won't (re-)mask the NFS services on upgrade, I think we have to keep the logic you have, to ensure we keep a masked service as masked after upgrade.